plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -104,15 +106,21 @@ export class SonataFlowService {
         ? `${args.serviceUrl}/${args.definitionId}?businessKey=${args.businessKey}`
         : `${args.serviceUrl}/${args.definitionId}`;
 
-      const result = await fetch(urlToFetch, {
+      const response = await fetch(urlToFetch, {
         method: 'POST',
         body: JSON.stringify(args.inputData),
         headers: { 'content-type': 'application/json' },
       });
 
-      const json = await result.json();
-      this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
-      return json;
+      if (response.ok) {
+        const json = await response.json();
+        this.logger.debug(`Execute workflow result: ${JSON.stringify(json)}`);
+        return json;
+      }
+      const res = await response.json();
+      throw new Error(
+        `${this.createPrefixFetchErrorMessage(urlToFetch, 'executeWorkflow', 'POST')} - Details: ${res?.details}, Stack: ${res?.stack}`,
+      );
     } catch (error) {
       this.logger.error(`Error when executing workflow: ${error}`);
```

Comment:
if we are already making this change let's also remove the try and catch
in case of network error it goes to the catch

Reply:
@gabriel-farache do you agree?

Reply:
I would prefer to keep the consistency across at least the file with those try-catch
Could this be in a follow-up PR/jira issue?

Reply:
anyhow it's not consistent now
I think better we have one example of proper error handling that we can refer to in future PRs